templates/pre-job.j2

```diff
@@ -36,11 +45,29 @@ elif [[ "${GITHUB_EVENT_NAME}" ==  "pull_request" ]]; then
   curl "${CURL_ARGS[@]}" \
       -d "{\"repository_name\": \"${GITHUB_REPOSITORY}\", \"source_repository_name\": \"${GITHUB_SOURCE_REPOSITORY}\", \"target_branch_name\": \"${GITHUB_BASE_REF}\", \"source_branch_name\": \"${GITHUB_HEAD_REF}\", \"commit_sha\": \"${GITHUB_SHA}\"}" \
       http://{{host_ip}}:8080/pull_request/check-run
+  REPO_CHECK=$?
 
 else
 
   logger -p user.error -s "${GITHUB_EVENT_NAME} is not supported yet. Please request it to be added on https://github.com/canonical/github-runner-operator/issues/new/choose"
 
-  return 1
+  REPO_CHECK=1
 
 fi
+
+if [[ $REPO_CHECK -ne 0 ]]; then
+  logger -p user.error -s "The repo-policy-compliance check failed with $REPO_CHECK exit code."
+  logger -p user.error -s "Stopping execution of jobs due to repo-policy-compliance failure."
+  logger -p user.error -s "This runner will be stopped or lost, fix the repo policy compliance then rerun this job."
+
+  # Killing the runner.Listener process to stop the runner application.
+  pkill -2 Runner.Listener
```

Comment:
Does this ensure that no further step from the job is processed?

Reply:
In this job run, I have tested with `if: always()` and `if: failure()`:
https://github.com/canonical/github-runner-operator/actions/runs/6033194405/job/16369693342

Both of step was not executed.